### ISSUE TITLE
Documentation on How to Create Lightweight Form

### ISSUE THREAD

**Issue by lienajp** [silverstripe/silverstripe-framework#5068]
Hi,

in the example mysite/templates/Includes/SearchForm.ss it would be good to add the security field because if one tries it like that he/she will get an error.

So the sentence 
`$Fields.dataFieldByName(SecurityID)`
 should be added to make the example working.

Idem dito for the error messages: I have been struggling for a while on how to display the error messages for fields and at form level.
So adding the following would be good as well.
1) error message at the level of the form:

```
    <% if $Message %>
        <p id="{$FormName}_error" class="message $MessageType">$Message</p>
    <% else %>
        <p id="{$FormName}_error" class="message $MessageType" style="display: none"></p>
    <% end_if %>

```

2) error message at the level of a field:

```
$Fields.dataFieldByName(Email)<span class="message $Fields.dataFieldByName(Email).messageType">$Fields.dataFieldByName(Email).Message()</span>
```

Thanks and best regards!
Philippe

**Comment by camfindlay** [comment]
Hi @lienajp - you can actually go ahead and update this documentation yourself if you have a working example. Simple click the edit button at the bottom of all our developer docs pages and it will allow you to quickly raise a pull request with your changes. Just make sure to prefix the commit message with "DOCS" so the core committers can find it easily. You might get some feedback and have the chance to revise anything before it gets added to the documentation.

Thanks.

**Comment by tractorcow** [comment]
SearchForm.php has security token disabled, which is why it's not included. see the `$this->disableSecurityToken();` in SearchForm.php

@lienajp just wondering what specific use case you had in mind for your custom templates. Typically the default form template suffices to show form and field-level error messages. I would start by copying the default templates if you needed to customise them.

Or maybe you're talking about a specific tutorial? :P Which is the example you're referencing? If you link the page you're viewing I'll double check the content is still correct.

**Comment by lienajp** [comment]
Hi, I am still developing the page. It's a form for the registration to events we organise. I wanted to add some Javascript/Jquery to the form to allow parts of it to be displayed or not depending if it is a member who wants to registrer or somebody new and to do other validations using JS.

It is easier to manipulate the layout of the form if I use my own template.

The reason why I proposed to add more info/examples in the documentation is that it took me 2 days to find out how I should do this. I used google to find out if anybody had the same problem and how they solved it. I find many people facing the same problem but no solutions. In fact, it's quite easy.

Once I can deploy it to production, I will give the link to the site and show the code. For the time being it works on my machine.

Best regards
Philippe
